src/atomate2/aims/jobs/convergence.py

```diff
@@ -0,0 +1,199 @@
+"""Defines the base FHI-aims convergence jobs."""
+
+from __future__ import annotations
+
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from jobflow import Flow, Maker, Response, job
+from pymatgen.core import Molecule, Structure
+
+from atomate2.aims.jobs.base import BaseAimsMaker
+from atomate2.aims.schemas.task import AimsTaskDoc, ConvergenceSummary
+from atomate2.aims.utils.msonable_atoms import MSONableAtoms
+
+CONVERGENCE_FILE_NAME = "convergence.json"  # make it a constant?
+
+
+@dataclass
+class ConvergenceMaker(Maker):
+    """Defines a convergence workflow with a maximum number of steps.
+
+    A job that performs convergence run for a given number of steps. Stops either
+    when all steps are done, or when the convergence criterion is reached, that is when
+    the absolute difference between the subsequent values of the convergence field is
+    less than a given epsilon.
+
+    Parameters
+    ----------
+    name : str
+        A name for the job
+    maker: .BaseAimsMaker
+        A maker for the run
+    criterion_name: str
+        A name for the convergence criterion. Must be in the run results
+    epsilon: float
+        A difference in criterion value for subsequent runs
+    convergence_field: str
+        An input parameter that changes to achieve convergence
+    convergence_steps: Iterable
+        An iterable of the possible values for the convergence field.
+        If the iterable is depleted and the convergence is not reached,
+        that the job is failed
+    last_idx: int
+        The index of the last allowed convergence step (max number of steps)
+    """
+
+    name: str = "convergence"
+    maker: BaseAimsMaker = field(default_factory=BaseAimsMaker)
+    criterion_name: str = "energy_per_atom"
+    epsilon: float = 0.001
+    convergence_field: str = field(default_factory=str)
```

Comment:
Does it make sense to have default values for this and `convergence_steps`? It seems like these need to be set to make this class useful.

Reply:
I put them there as the most common use case for the class - tighten the inputs until the energy per atom converges up to 1 meV/atom. However, it makes sense also to remove them completely

Reply:
Sorry for the confusion. I wasn't talking about `criterion_name` or `epsilon`. I just meant that `convergence field` and `convergence_steps` have default values as `field(default_factory=str)` etc. This means that you can technically initialise this class without setting a value for these fields. It could be better to remove the `= field(...)` parts entirely to make sure the user sets the correctly.

Reply:
ah, ok. Yes, these should be mandatory